pkg/operator/configobservation/auth/webhook_authenticator.go

```diff
@@ -45,8 +123,21 @@ func ObserveWebhookTokenAuthenticator(genericListers configobserver.Listers, rec
 		// keep going on read error from existing config
 		errs = append(errs, err)
 	}
-	existingWebhookConfigured := len(existingWebhookAuthenticator) > 0
 
+	if isReady, err := IsWebhookAuthenticatorReady(listers, existingConfig); err != nil {
+		// if we had an error determining, just show the error and keep the existing config.
+		return existingConfig, append(errs, err)
+	} else if !isReady {
+		// if we aren't yet ready to use the webhook configuration, clear the webhook configuration
+		// don't sync anything and remove whatever we synced
+		resourceSyncer.SyncSecret(
+			resourcesynccontroller.ResourceLocation{Namespace: operatorclient.TargetNamespace, Name: "webhook-authenticator"},
+			resourcesynccontroller.ResourceLocation{Namespace: "", Name: ""},
+		)
+		return map[string]interface{}{}, nil
+	}
+
+	existingWebhookConfigured := len(existingWebhookAuthenticator) > 0
```

Comment:
I'd rather see this handled on the CAO side where the authenticators wouldn't be set until oauth-apiserver is ready